WSLDVCPlugin/WSLDVCCallback.cpp

```diff
@@ -173,7 +174,10 @@ class WSLDVCCallback :
         ReadUINT32(iconData->iconFormat, cur, len);
         ReadUINT32(iconData->iconBitsLength, cur, len);
 
-        iconData->iconFileSize = sizeof(ICON_HEADER) + sizeof(BITMAPINFOHEADER) + iconData->iconBitsLength;
+        if (UIntAdd(sizeof(ICON_HEADER) + sizeof(BITMAPINFOHEADER), iconData->iconBitsLength, &iconData->iconFileSize) != S_OK) {

```

Comment:
Mnior: UIntAdd already returns an HRESULT, use that return value for a more accurate error return.  For example:

```
HRESULT hr = UIntAdd(sizeof(ICON_HEADER) + sizeof(BITMAPINFOHEADER), iconData->iconBitsLength, &iconData->iconFileSize);
if (FAILED(hr)) {
    return hr;
}
```